base/path.jl

```diff
@@ -611,3 +611,21 @@ relpath(path::AbstractString, startpath::AbstractString) =
 for f in (:isdirpath, :splitdir, :splitdrive, :splitext, :normpath, :abspath)
     @eval $f(path::AbstractString) = $f(String(path))
 end
+
+"""
+    uripath(path::AbstractString)
+
+Encode `path` as a URI as per RFC1738, RFC3986, and the
+[Freedesktop File URI spec](https://www.freedesktop.org/wiki/Specifications/file-uri-spec/).
+"""
+function uripath(path::AbstractString)
+    percent_escape(s) = '%' * join(map(
+        b -> uppercase(string(b, base=16)),
```

Comment:
I don't think this is usually uppercased
```suggestion
        b -> string(b, base=16),
```